helm-plugins/snapshot/main.go

```diff
@@ -0,0 +1,244 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/layer5io/meshkit/logger"
+)
+
+var log logger.Handler
+
+// Config holds the configuration data passed during build or runtime
+type Config struct {
```

Comment:
An aspect of this plugin that would be nice to enhance is its use of the user’s potentially available local meshconfig found in `~/.meshery/config.yaml`.  in this way, a number of these environment values can be directly derived from their existing Meshery configuration file.